pkgs/development/tools/build-managers/bazel/bazel_3/default.nix

```
@@ -507,9 +507,15 @@ stdenv.mkDerivation rec {
     mv ./bazel_src/output/bazel $out/bin/bazel-${version}-${system}-${arch}
 
     # shell completion files
-    mkdir -p $out/share/bash-completion/completions $out/share/zsh/site-functions
-    mv ./bazel_src/output/bazel-complete.bash $out/share/bash-completion/completions/bazel
-    cp ./bazel_src/scripts/zsh_completion/_bazel $out/share/zsh/site-functions/
+    install -Dm644 \
+      ./bazel_src/output/bazel-complete.bash \
+      $out/share/bash-completion/completions/bazel/bazel-complete.bash
+    install -Dm644 \
+      ./bazel_src/scripts/zsh_completion/_bazel \
+      $out/share/zsh/site-functions/_bazel
+    install -Dm644 \
+      ./bazel_src/scripts/fish/completions/bazel.fish \
+      $out/share/fish/vendor_completions.d/bazel.fish
```

```suggestion
    installShellCompletion bazel_src/output/bazel-complete.bash
    installShellCompletion bazel_src/scripts/fish/completions/bazel.fish
    installShellCompletion --zsh bazel_src/scripts/zsh_completion/_bazel
```

Should do the trick with `installShellFiles` in `nativeBuildInputs`.

---

Great! I always forget that this exists.